### PR TITLE
core: small refactoring on job.py and a smaller run_suite() signature

### PR DESCRIPTION
**Pull request by beraldoleal**
With the progress that we made on Avocado's config dict we can now pass the entire dict to methods and avoid long method signatures.

This is part of bigger refactoring to allow run jobs with multiple suites.

### CODE REVIEW

File: avocado/plugins/runner_nrunner.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def _populate_task_logdir(self, base_path, task, statuses, debug=False):
         with open(data_file, 'w') as fp:
             fp.write("{}\n".format(task.output_dir))
 
-    def run_suite(self, job, result, test_suite, variants, timeout=0,
-                  replay_map=None, execution_order=None):
+    def run_suite(self, job, result, test_suite, variants, config):
         summary = set()
+        timeout = config.get('run.job_timeout')
```
Comment (willianrampazzo) [inline avocado/plugins/runner_nrunner.py:77]: Don't we need to check the time format, just as you did on `avocado/plugins/runner.py`?

`timeout = data_structures.time_to_seconds(config.get('run.job_timeout'))`

File: avocado/core/job.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
                loader, nrunner, output, result, tags, varianter, version)
 from .future.settings import settings
 from .output import LOG_JOB, LOG_UI, STD_OUTPUT
+from .resolver import (resolve, ReferenceResolutionResult)
```
Comment (clebergnu) [inline avocado/core/job.py:36]: Do you suggest we switch to this import style? I mean, as it stands, it's confusing for people trying to contribute if they mean to follow a given style.

File: avocado/core/job.py [hunk 3]
```diff
@@ -434,14 +436,14 @@ def _resolver_check_missing_references(references, resolutions):
...
-            if resolver.ReferenceResolutionResult.SUCCESS not in results:
+            if ReferenceResolutionResult.SUCCESS not in results:
                 missing.append(reference)
         if missing:
             msg = "Could not resolve references: %s" % ",".join(missing)
             raise exceptions.JobTestSuiteReferenceResolutionError(msg)
 
     def _make_test_suite_resolver(self, references, ignore_missing):
-        resolutions = resolver.resolve(references)
+        resolutions = resolve(references)
```
Comment (clebergnu) [inline avocado/core/job.py:446]: This is not related to the commit message, or am I missing something?

File: avocado/core/job.py (hunk 2 above)
Comment (willianrampazzo) [reply avocado/core/job.py:36]: Looking one more time to it, I agree with @clebergnu; we should define a standard and follow it.

Comment (beraldoleal) [reply avocado/core/job.py:36]: According to PEP8 is ok to import like that, this was also a result of a `isort` command that automatically replaced this. I'm ok with both approaches here.

Comment (willianrampazzo) [reply avocado/core/job.py:36]: I'm also okay with any of the approaches, but I think we should define and follow one of them as our standard. It is just a matter of making a choice here as the style we will follow.

Comment (beraldoleal) [reply avocado/core/job.py:36]: Both styles are accepted by PEP8 and will not break any linter. I prefer this way because:

1. `isort` does the job automatically and;
2. reduces the number of lines

Comment (beraldoleal) [reply avocado/core/job.py:36]: Since that the comment was here and not on the previous commit, I'm assuming that maybe the () might be strange here. And it was added by mistake. So, is this better?

`from .resolver import ReferenceResolutionResult,  resolve`

I also did this to reduce the length of the lines at the code when using this and also to avoid import the entire module when we are using only one method and one enum.

File: avocado/core/job.py (hunk 3 above)
Comment (beraldoleal) [reply avocado/core/job.py:446]: @clebergnu because of the changes in the loop and how I replaced the `if resolver.ReferenceResolutionResult.SUCCESS` to fit on 80 columns I decided to rewrite the import. Because of that I changed the only other place that uses `resolver`, here.

Makes sense?

File: avocado/core/job.py (hunk 2 above)
Comment (willianrampazzo) [reply avocado/core/job.py:36]: I went true the PEP8 about _imports_ and I like the approach:

`from package import foo, bar, foz, baz`

I'm in about using this style from now on. But again, I think we should decide to go with one or another, even if PEP8 mention both are fine.

File: avocado/plugins/runner_nrunner.py (hunk 1 above)
Comment (beraldoleal) [reply avocado/plugins/runner_nrunner.py:77]: Nice catch, thanks.

But after a better thinking, maybe the best approach is to register_option() with `key_type=time_to_seconds`. Will fix on a v2.

**Comment by beraldoleal** [comment]
I'm preparing a v2.

File: avocado/core/job.py (hunk 3 above)
Comment (clebergnu) [reply avocado/core/job.py:446]: > @clebergnu because of the changes in the loop and how I replaced the `if resolver.ReferenceResolutionResult.SUCCESS` to fit on 80 columns I decided to rewrite the import. Because of that I changed the only other place that uses `resolver`, here.
> 
> Makes sense?

That was my point... I was missing the other motivation ("fit 80 columns").  So maybe just a note on the commit message to explain the why of this change ("while at it, fit to 80 columns").
